real-main/app/models/user/model.py

```diff
@@ -350,6 +356,11 @@ def start_change_contact_attribute(self, attribute_name, attribute_value):
             names['cognito']: attribute_value,
             f'custom:unverified_{names["short"]}': attribute_value,
         }
+
+        # verify that new attribtue value is not used by other
+        if self.email_dynamo.get(attribute_value) or self.phone_number_dynamo.get(attribute_value):
```

Comment:
This works but as I imagine you know, it will query email addresses for a phone number, which is a waste of a query. Here's one way to avoid that useless query.

First, change the definition of `CONTACT_ATTRIBUTE_NAMES` like so:

```py
CONTACT_ATTRIBUTE_NAMES = {
    'email': {
        'short': 'email',
        'cognito': 'email',
        'dynamo_attr': 'email',
        'dynamo_client': 'email_dynamo',
    },
    'phone': {
        'short': 'phone',
        'cognito': 'phone_number',
        'dynamo_attr': 'phoneNumber',
        'dynamo_client': 'phone_number_dynamo',
    },
}
```

then this part here becomes

```suggestion
        contact_attr_dyamo = getattr(self, names[attribute_name])
        if contact_attr_dynamo.get(attribute_value):
```

Also, existing code of `names['dynamo']` will need to be changed to `names['dynamo_attr']`